### PR TITLE
[MOBILE APP] update screen

### PR DESCRIPTION
**Pull request by NoahNsimbe**
#### Summary of Changes (What does this PR do?)

- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?
[MOB-184](https://airqoteam.atlassian.net/browse/MOB-184) 

#### Screenshots (optional)
Screenshot_1677063219

[MOB-184]: https://airqoteam.atlassian.net/browse/MOB-184?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mobile/lib/screens/home_page.dart [hunk 1]
```diff
@@ -194,6 +195,17 @@ class _HomePageState extends State<HomePage> {
...
+      if (context.read<DashboardBloc>().state.checkForUpdates) {
+        await AppService().latestVersion().then((version) async {
+          if (version != null && mounted) {
+            await canLaunchUrl(version.url).then((bool result) async {
+              await openUpdateScreen(context, version);
+            });
+          }
+        });
+      }
+    });
```
Comment (NoahNsimbe) [inline mobile/lib/screens/home_page.dart:208]: Showing update screen on home screen

File: mobile/lib/screens/on_boarding/introduction_screen.dart [hunk 2]
```diff
@@ -95,6 +98,17 @@ class IntroductionScreenState extends State<IntroductionScreen> {
...
+      if (context.read<DashboardBloc>().state.checkForUpdates) {
+        await AppService().latestVersion().then((version) async {
+          if (version != null && mounted) {
+            await canLaunchUrl(version.url).then((bool result) async {
+              await openUpdateScreen(context, version);
+            });
+          }
+        });
+      }
+    });
```
Comment (NoahNsimbe) [inline mobile/lib/screens/on_boarding/introduction_screen.dart:111]: Showing update screen on introduction screen

File: mobile/lib/utils/extensions.dart [hunk 3]
```diff
@@ -609,6 +609,41 @@ extension FileExt on File {
...
+    // checking second code
+    if (versionSections[1] > candidateSections[1]) return 1;
+
+    if (versionSections[1] < candidateSections[1]) return -1;
+
+    // checking last code
+    if (versionSections.last > candidateSections.last) return 1;
+
+    if (versionSections.last < candidateSections.last) return -1;
+
```
Comment (NoahNsimbe) [inline mobile/lib/utils/extensions.dart:642]: Comparing versions

File: mobile/test/extensions_test.dart [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import 'package:app/models/models.dart';
+import 'package:app/utils/utils.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  group('AppStoreVersion', () {
+    test('store version should be greater than user version', () {
+      const String userAppVersion = '1.0.0';
```
Comment (NoahNsimbe) [inline mobile/test/extensions_test.dart:8]: tests to comparing versions

**Review by Baalmart — APPROVED** [review]
Thanks @NoahNsimbe , this looks fine on my end.

I was supposed to add more feedback notes but they have escaped my mind due to over multitasking.

Thanks again!

**Review by BenjaminSsempala — APPROVED** [review]
LGTM

**Pull request merged**